src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -445,7 +445,25 @@ export default class ReimbursementRequestService {
    * @returns the created vendor
    */
   static async createVendor(submitter: User, name: string) {
-    if (!isAdmin(submitter.role)) throw new AccessDeniedAdminOnlyException('create vendors');
+    /* AUTHORIZATION CHECK */
+    const failedAuthorizationException = new AccessDeniedException(
+      'Only admins, finance leads, and finance heads can create vendors.'
+    );
+
+    if (!process.env.FINANCE_TEAM_ID) throw failedAuthorizationException;
+
+    // Check finance team manually because helpers in reimbursement-requests.utils.ts don't work as desired
+    const financeTeam = await prisma.team.findUnique({
+      where: { teamId: process.env.FINANCE_TEAM_ID },
+      include: { head: true, leads: true, members: true }
+    });
+
+    if (!financeTeam) throw failedAuthorizationException;
+
+    const authorized =
+      isAdmin(submitter.role) || submitter.userId === financeTeam.headId || financeTeam.leads.includes(submitter);
+    if (!authorized) throw failedAuthorizationException;
+    /* END AUTHORIZATION CHECK */
```

Comment:
so all of this finance team logic should be in a helper method. I think we should change validateUserIsPartOfFinanceTeam to return a boolean instead of throwing an error, that way we can use it in tandum with other permissions checks